### PR TITLE
Small fixes to job submission cluster docs

### PR DESCRIPTION
**Pull request by cadedaniel**
## Why are these changes needed?
I walked through the new job submission cluster docs and sanded down a few rough edges.

### CODE REVIEW

File: doc/source/cluster/running-applications/job-submission/quickstart.rst [hunk 1]
```diff
@@ -106,6 +106,11 @@ To submit the Ray Job, we use ``ray job submit``:
 
 By default, this command will run the script on the Ray Cluster and wait until the job has finished. Also note the ``hello world`` printout: the stdout of the job is streamed back to the client.
 
+.. note::
```
Comment (stephanie-wang) [inline doc/source/cluster/running-applications/job-submission/quickstart.rst:109]: Could we actually just use this command in the example? I think that will be the least confusing, so there aren't two different versions of the command.

Also, can you include a brief explanation of what the argument does under the hood?

**Review by stephanie-wang — CHANGES_REQUESTED** [review]
Let's also update the preceding instructions to create an empty working directory and create the script there (when I first tried to follow the instructions from the ray repo directory, I got an error that the working directory was too big).

File: doc/source/cluster/running-applications/job-submission/rest.rst [hunk 2]
```diff
@@ -32,7 +33,7 @@ Under the hood, both the Python SDK and the CLI make HTTP calls to the job serve
     start = time.time()
     while time.time() - start <= 10:
         resp = requests.get(
-            "http://127.0.0.1:8265/api/jobs/<job_id>"
+            f"http://127.0.0.1:8265/api/jobs/{job_id}"
```
Comment (stephanie-wang) [inline doc/source/cluster/running-applications/job-submission/rest.rst:36]: Nice change!

File: doc/source/cluster/running-applications/job-submission/quickstart.rst (hunk 1 above)
Comment (cadedaniel) [reply doc/source/cluster/running-applications/job-submission/quickstart.rst]: Done, LMK what you think.

File: doc/source/cluster/running-applications/job-submission/quickstart.rst [hunk 3]
```diff
@@ -104,7 +105,9 @@ To submit the Ray Job, we use ``ray job submit``:
     # Job 'raysubmit_inB2ViQuE29aZRJ5' succeeded
     # ------------------------------------------
 
-By default, this command will run the script on the Ray Cluster and wait until the job has finished. Also note the ``hello world`` printout: the stdout of the job is streamed back to the client.
+This command will run the script on the Ray Cluster and wait until the job has finished. Note that it also streams the stdout of the job back to the client (``hello world`` in this case).
```
Comment (stephanie-wang) [inline doc/source/cluster/running-applications/job-submission/quickstart.rst:108]:
```suggestion
This command will run the script on the Ray Cluster and wait until the job has finished. Note that it also streams the stdout of the job back to the client (``hello world`` in this case). Ray will also make the contents of the directory passed as `--working-dir` available to the Ray job by downloading the directory to all nodes in your cluster.
```

**Review by stephanie-wang — APPROVED** [review]

File: doc/source/cluster/running-applications/job-submission/quickstart.rst [hunk 4]
```diff
@@ -104,7 +105,9 @@ To submit the Ray Job, we use ``ray job submit``:
     # Job 'raysubmit_inB2ViQuE29aZRJ5' succeeded
     # ------------------------------------------
 
-By default, this command will run the script on the Ray Cluster and wait until the job has finished. Also note the ``hello world`` printout: the stdout of the job is streamed back to the client.
+This command will run the script on the Ray Cluster and wait until the job has finished. Note that it also streams the stdout of the job back to the client (``hello world`` in this case). Ray will also make the contents of the directory passed as `--working-dir` available to the Ray job by downloading the directory to all nodes in your cluster.
+
+Ray will automatically make the contents of the working directory available to the Ray job. This enables you to focus on iterating on your Ray app instead of worrying about packaging.
```
Comment (cadedaniel) [inline doc/source/cluster/running-applications/job-submission/quickstart.rst:110]: @stephanie-wang looks a little repetitive with your edit. Want me to remove this second paragraph?

Comment (stephanie-wang) [reply doc/source/cluster/running-applications/job-submission/quickstart.rst:110]: Ah thanks, meant to remove this part.

Comment (stephanie-wang) [inline doc/source/cluster/running-applications/job-submission/quickstart.rst:110]:
```suggestion
```

**Pull request merged**